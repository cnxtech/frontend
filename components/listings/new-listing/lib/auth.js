import { getUserInfo, getPhoneParts } from 'lib/user'

const getUser = async (id, updatePhone) => {
  const userInfo = await getUserInfo(id)
  if (userInfo && !userInfo.error) {
    // Update user info in redux
    const fullPhone = getPhoneParts(userInfo.phone)
    const name = userInfo.name
    updatePhone({
      name: name,
      internationalCode: fullPhone.internationalCode,
      localAreaCode: fullPhone.localAreaCode,
      number: fullPhone.number
    })
  }

  return userInfo
}

const hasPhoneNumber = (phone) => {
  if (!phone) {
    return false
  }
  const { internationalCode, localAreaCode, number } = phone
  return (
    internationalCode !== null &&
    localAreaCode !== null &&
    number !== null
  )
}

export {
  getUser,
  hasPhoneNumber
}
