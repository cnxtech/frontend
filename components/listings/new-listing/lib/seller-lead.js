import {captureException} from '@sentry/browser'
import moment from 'moment'
import {CREATE_LEAD, TOUR_SCHEDULE} from 'graphql/listings/mutations'
import {getSellerLeadInput} from 'lib/listings/insert'
import {
  log,
  SELLER_ONBOARDING_LISTING_CREATION_SUCCESS,
  SELLER_ONBOARDING_LISTING_CREATION_ERROR,
  SELLER_ONBOARDING_TOUR_CREATION_SUCCESS,
  SELLER_ONBOARDING_TOUR_CREATION_ERROR
} from 'lib/logging'

/**
 * Calls the mutation to create a tour. Returns true if success, false otherwise.
 */
export const requestCreateTour = async (apolloClient, {day, time, uuid}) => {
  const datetime = moment(day + time, 'YYYY-MM-DD HH').toDate()
  try {
    const { data } = await apolloClient.mutate({
      mutation: TOUR_SCHEDULE,
      variables: {
        input: {
          siteSellerLeadUuid: uuid,
          options: {
            datetime
          },
          wantsTour: true,
          wantsPictures: true
        }
      }
    })

    if (data) {
      log(SELLER_ONBOARDING_TOUR_CREATION_SUCCESS, {
        uuid: uuid,
        options: datetime
      })
      return true
    }
  } catch (e) {
    captureException(e)
    log(SELLER_ONBOARDING_TOUR_CREATION_ERROR, {
      uuid: uuid,
      options: datetime,
      error: e && e.message ? e.message : ''
    })
  }
  return false
}

/**
 * Calls the mutation to create a seller lead. Returns the lead uuid if success,
 * null otherwise.
 */
export const requestCreateLead = async (apolloClient, listingData) => {
  const input = getSellerLeadInput(listingData)
  try {
    const { data } = await apolloClient.mutate({
      mutation: CREATE_LEAD,
      variables: {
        input
      }
    })

    if (data) {
      log(SELLER_ONBOARDING_LISTING_CREATION_SUCCESS, {listing: input})
      return data.siteSellerLeadCreate.uuid
    }
  } catch (e) {
    captureException(e)
    log(SELLER_ONBOARDING_LISTING_CREATION_ERROR, {
      listing: input,
      error: e && e.message ? e.message : ''
    })
  }
  return null
}
