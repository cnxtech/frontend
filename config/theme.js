import theme from '@emcasa/ui'
import {fas} from '@fortawesome/pro-solid-svg-icons'
import {fal} from '@fortawesome/pro-light-svg-icons'
import icons from './icons'

export default {
  ...theme,
  icons: {emcasa: icons, default: fas, light: fal}
}
