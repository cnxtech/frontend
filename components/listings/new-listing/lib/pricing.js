import {captureException} from '@sentry/browser'
import {getAddressInput} from 'lib/address'
import {estimatePrice, getPricingInput} from 'lib/listings/pricing'
import {
  log,
  getSellerEventPrefix,
  SELLER_ONBOARDING_PRICING_SUCCESS,
  SELLER_ONBOARDING_PRICING_FAILED
} from 'lib/logging'

export const requestPricing = async (apolloClient, userInfo, {homeDetails, rooms, location, pricing, updatePricing, evaluation}) => {
  // Prepare input
  const addressInput = getAddressInput(location.addressData)
  const pricingInput = getPricingInput(addressInput, homeDetails, rooms, userInfo)

  // Run mutation
  const response = await estimatePrice(apolloClient, pricingInput)
  if (response.error) {
    captureException(new Error(response.error))
    return response
  }

  // Handle result
  if (response.result) {
    const {id, suggestedPrice, userPrice} = response.result
    if (suggestedPrice) {
      log(`${getSellerEventPrefix(evaluation)}${SELLER_ONBOARDING_PRICING_SUCCESS}`, {
        name: userInfo.name,
        phone: userInfo.phone,
        priceRequestId: id,
        pricingInput,
        suggestedPrice: suggestedPrice
      })
    } else {
      log(`${getSellerEventPrefix(evaluation)}${SELLER_ONBOARDING_PRICING_FAILED}`, {
        name: userInfo.name,
        phone: userInfo.phone,
        pricingInput
      })
    }
    updatePricing({
      ...pricing,
      priceRequestId: id,
      suggestedPrice,
      userPrice
    })
  } else {
    return {
      error: 'No response.'
    }
  }
}