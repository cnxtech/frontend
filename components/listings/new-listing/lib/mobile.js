import theme from '@emcasa/ui'

const MOBILE_BREAKPOINT_WIDTH = parseInt(theme.breakpoints[0])

const isMobile = () => {
  if (!process.browser) {
    return false
  }
  return screen.width <= MOBILE_BREAKPOINT_WIDTH
}

export {
  isMobile
}
