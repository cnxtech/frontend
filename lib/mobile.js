import theme from '@emcasa/ui'

/**
 * @deprecated. This method ends up causing a difference between what is rendered on the server and what is rendered on
 * the client. We should replace this with <Mobile> and <Desktop> components.
 */
const isMobile = (width) => {
  if (!process.browser) {
    return false
  }
  return (width || window.innerWidth) <= parseInt(theme.breakpoints[1])
}

export {
  isMobile
}
