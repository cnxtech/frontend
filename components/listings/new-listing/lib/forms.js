import {isMobile} from 'lib/mobile'

const autoFocus = (ref) => {
  if (ref && !isMobile()) {
    ref.focus()
  }
}

export {
  autoFocus
}
