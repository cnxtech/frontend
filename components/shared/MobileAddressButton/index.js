import React, { PureComponent } from 'react'
import PropTypes from 'prop-types'
import { isMobile } from 'lib/mobile'
import {
  Container,
  MobileAddressTextContainer,
  MobileAddressText
} from './styles'

class MobileAddressButton extends PureComponent {
  render() {
    const { address, onClick } = this.props

    return (
      <Container
        fluid
        height={isMobile() ? 'tall' : 'medium'}
        onClick={onClick}>
        <MobileAddressTextContainer>
          <MobileAddressText
            inline
            hasAddress={address !== null}
          >
            {address ? address : 'Endereço e número*'}
          </MobileAddressText>
        </MobileAddressTextContainer>
      </Container>
    )
  }
}

MobileAddressButton.propTypes = {
  onClick: PropTypes.func.isRequired,
  address: PropTypes.string
}

export default MobileAddressButton
