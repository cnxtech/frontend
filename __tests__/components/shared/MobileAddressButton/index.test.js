import React from 'react'
import MobileAddressButton from 'components/shared/MobileAddressButton'
import renderer from 'react-test-renderer'
import 'jest-styled-components'
import theme from '@emcasa/ui'
import { ThemeProvider } from 'styled-components'

import { isMobile } from 'lib/mobile'
jest.mock('lib/mobile')
isMobile.mockImplementation(() => false)

describe('<MobileAddressButton/>', () => {
  it('should render component on desktop', () => {
    const tree = renderer.create(
      <ThemeProvider theme={theme}>
        <MobileAddressButton onClick={() => {}} />
      </ThemeProvider>).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
