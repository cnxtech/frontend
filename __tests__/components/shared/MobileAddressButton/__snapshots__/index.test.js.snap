// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MobileAddressButton/> should render component on desktop 1`] = `
.c0 {
  outline: none;
  display: block;
  cursor: pointer;
  padding: 0 20px 0 20px;
  border-color: #e0e6ed;
  background-color: #ffffff;
  border-radius: 4px;
  border-width: 1px;
  width: 100%;
  overflow: hidden;
  text-overflow: ellipsis;
  white-space: nowrap;
  height: 40px;
  color: #3c4858;
  font-family: Rubik;
  font-size: 18px;
  -webkit-font-smoothing: antialiased;
  border-style: solid;
  font-size: 18px;
  text-align: left;
  padding: 0 10px 0 10px;
}

.c0 :hover {
  background-color: #ffffff;
}

.c2 {
  -webkit-font-smoothing: antialiased;
  font-family: Rubik;
  color: #3c4858;
  font-size: 18px;
  white-space: nowrap;
  color: #3c4858;
}

.c1 {
  overflow: hidden;
  text-overflow: ellipsis;
}

<button
  className="c0"
  fontSize="default"
  height="medium"
  onClick={[Function]}
>
  <div
    className="c1"
  >
    <span
      className="c2"
      color="dark"
      fontSize="default"
      hasAddress={true}
    >
      Endereço e número*
    </span>
  </div>
</button>
`;
